### ISSUE TITLE
Doesn't work when using the "proxy" config

### ISSUE THREAD

**Issue by maximedupre** [johnpapa/lite-server#68]
Here are my configs:

    {
    	"proxy": "localhost:5000",
    	"port": 7000
    }

When I run the command **without** `"proxy": "localhost:5000"`, an instance of Chrome opens on `localhost:7000` as expected and everything is fine. **With** `"proxy": "localhost:5000"`, it shows:

    ** browser-sync config **
    { injectChanges: false,
      files: [ './**/*.{html,htm,css,js}' ],
      watchOptions: { ignored: 'node_modules' },
      server: { baseDir: './', middleware: [ [Function], [Function] ] },
      logLevel: 'silent',
      proxy: 'localhost:5000',
      port: 7000 }

in the terminal and the process exits.

I tested a bit with some `console.log`s  in `lite-server.js`:

    console.log('executed');
    browserSync.init(config, cb);
    console.log('not executed');

Versions:
- `lite-server` version: 2.2.0
- `nodejs` version: 5.9.1
- `npm` version: 3.7.3
- OS type/version: OSX El Capitan 10.11.3

**Update**
I gave version `2.1.0` a try and this time I got much clearer indications: `Invalid config. You cannot specify both server & proxy options`.

When merging the default config and the custom config, `proxy` and `server` are put together in the same set of configurations.

To temporarily fix my problem, I replaced

    if (config.server.middleware) {
        config.server.middleware = _.compact(config.server.middleware);
    }

by:

    delete config.server;

So there are two problems:
1. The outputs from `browser-sync` are not being displayed in the terminal (They are displayed in 2.1.0. but not 2.2.0). This includes `Invalid config. You cannot specify both server & proxy options`, but also the message telling me to add

    <script id="__bs_script__">//<![CDATA[
    //     document.write("<script async src='http://HOST:3000/browser-sync/browser-sync-    client.2.11.2.js'><\/script>".replace("HOST", location.hostname));
    // //]]></script>

before the closing tag of the `body` when the application actually launches in proxy mode.
2. There is a conflict between the default `server` config and the `proxy` config.

**Comment by cgmartin** [comment]
The `proxy` and `server` configs in browsersync are mutually exclusive. The current lite-server always uses `server` in the config, so browsersync fails when including `proxy` config.

When trying the following `bs-config.json` with Browsersync (without lite-server), it proxies as expected:
```
{ "proxy": "http://www.example.com/" }
$ browser-sync --config bs-config.json start
```

But as soon as you add `server` config, it fails with an error:
```
{ 
  "proxy": "http://www.example.com/",
  "server": { "baseDir": "./" } 
}
$ browser-sync --config bs-config.json start
Invalid config. You cannot specify both server & proxy options.
```

@maximedupre could you describe your use-case a bit more? Curious how beneficial it would be for lite-server to support proxy mode vs. using browsersync itself.

**Comment by maximedupre** [comment]
@cgmartin I was updating my initial post when you submitted your comment. Please see the update.

**Comment by cgmartin** [comment]
1. A fix should be possible for the lack of error logging. Good catch.
2. could you describe your use-case a bit more? Curious how beneficial it would be for lite-server to support proxy mode vs. using browsersync itself.

**Comment by maximedupre** [comment]
I am working on a SPA and my comprehension of `lite-server` is that it works well with SPAs, as opposed to `browser-sync`.

The reason I need to use a proxy is because I am developing the front and back ends at the same time. Since the (nodejs) server and `lite-server` can't have the same port, I need to use the nodejs server port as a proxy to `lite-server`. Essentially, I am doing this: https://gist.github.com/sogko/b53d33d4f3b40d3b4b2e

**Comment by cgmartin** [comment]
Makes sense. Would either of these alternative options be viable for your use-case?

* Use `lite-server` as the static file server, and proxy via middleware to the separate nodejs back-end server on different port (see [example](https://github.com/johnpapa/lite-server/issues/61#issuecomment-205997607)).
* Replace `lite-server` with your own server that handles both static file serving and AJAX API requests, and use `browser-sync` in proxy mode for development. The `index.html` fallback functionality from `lite-server` would just need to exist in your static file server.

**Comment by maximedupre** [comment]
I don't seem to get alternative number 1 to work. This is my middleware:

    1: proxyMiddleware('/', {
	    target: 'localhost:5000'
    })

(My nodejs server handles both static file serving and AJAX API requests, which is why I redirect everything to localhost:5000).

When I start `lite-server`, it opens the browser and gives me this message: 
> Error occured while trying to proxy to: localhost:3000/

In the terminal:
> [HPM] PROXY ERROR: EHOSTUNREACH. localhost:3000 -> localhost:5000/
16.04.10 17:59:28 500 GET /

However, I can access localhost:5000 just fine when I try manually. Maybe I'm doing something wrong, but I haven't figured it out yet.

As for alternative number 2, since `browser-sync` doesn't handle SPAs, wouldn't I lose most of the benefits?

Thank you for your help.

**Comment by cgmartin** [comment]
Glad it is working! BTW, there were some recent issues with proxy mode that are fixed in Browsersync `2.12.2+` (see #71). 

To remove the request logging, there's an example in the README:
> Another example: To remove one of the default middlewares, such as connect-logger, you can set it's array index to null:
```
module.exports = {
  server: {
    middleware: {      
      0: null     // removes default `connect-logger` middleware
    }
  }
};
```

**Comment by maximedupre** [comment]
Ah right! Adding `logLevel: 'silent'` actually works, but it's removing a different layer of logs. Setting the first middleware to 0 solves the problem. Thanks!
